scripts/build/TestImpactAnalysis/persistent_storage/tiaf_persistent_storage_local.py

```
@@ -108,9 +108,8 @@ def _copy_files_from_directory_to_destination(self, source_directory: pathlib.Pa
         @param source_directory: pathlib.Path to directory to copy files from.
         @param target_direcotry: pathlib.Path to directory to store files in.
         """
-        for artifact_path in source_directory.iterdir():
-            try:
-                shutil.copy2(artifact_path, target_directory.joinpath(artifact_path.name))
-            except OSError as e:
-                logger.error(f"Error copying file {artifact_path.name} from {source_directory} to {target_directory}")
+        try:
+            shutil.copytree(source_directory, target_directory, dirs_exist_ok=True)
+        except OSError as e:
+                logger.error(f"Error copying tree '{source_directory}' to '{target_directory}'")
```

nit: Since this is being updated anyways, can you report the exception/error on the same logger.error instead of doing it in a separate error logger call? (and remove the separate logger below this line)

---

logger.exception is a handy way to log an error and automatically include stacktrace